python/cudf/cudf/core/dataframe.py

```diff
@@ -1876,7 +1891,22 @@ def _make_operands_and_index_for_binop(
             for k, v in rhs.items():
                 if k not in lhs:
                     operands[k] = (left_default, v, reflect, None)
-        return operands, index
+
+        if cudf.get_option("mode.pandas_compatible") and not equal_columns:
+            if isinstance(other, DataFrame):
+                column_names_list = self._data.to_pandas_index().join(
+                    other._data.to_pandas_index(), how="outer"
+                )
+            elif isinstance(other, Series):
+                column_names_list = self._data.to_pandas_index().join(
+                    other.index.to_pandas(), how="outer"
+                )
```

Comment:
Does this need an "else raise"? I don't want to run into an unbound variable error.
```suggestion
                )
            else:
                raise ValueError("other must be a DataFrame or Series.")
```